### PR TITLE
Kanto Ctl helper

### PR DESCRIPTION
**Pull request by mikehaller**
This PR brings a new shell script to manage Kanto Container Manager configuration:
- Add and Remove "Container Registry Configuration" entries, e.g. authentication credentials for private container registries
- Set arbitrary primitives in the configuration file

The script `sdv-kanto-ctl` as a help. It creates backups of the original configuration file and prints the applied changes.

```
$ ./sdv-kanto-ctl --help
Eclipse Kanto Container Manager Configuration Utility
See https://websites.eclipseprojects.io/kanto/docs/references/containers/container-manager-config/
Usage: ./sdv-kanto-ctl <command> {options}
Commands:
        add-registry -h <hostname> -u <username> -p <password>
                Adds or replaces a container registry authentication configuration
                -h or --hostname: Configure the hostname of the container registry (e.g. hub.docker.io, ghcr.io, ...)
                -u or --username: Configure the username
                -p or --password: Configure the password
        remove-registry -h <hostname>
                Removes the specified container registry
                -h or --hostname: The hostname of the container registry
        remove-all-registries
                Removes all configured container registries
        list-registries
                Prints all configured container registries
        show-config
                Print the container management configuration
        set <key> <value>
                Set a primitive configuration value. Key in JSON Dot-Notation
                Examples: ./sdv-kanto-ctl set containers.registry_configurations.MyRegistry.credentials.password foobar
                          ./sdv-kanto-ctl set things.enable true
Options:
        --no-reload : Do not reload the configuration and restart the container-management service automatically
        --ansi : Don't use colored output.
        --verbose | -v : Enable verbose mode.
        --help : This message.
```

### CODE REVIEW

File: src/sh/sdv-kanto-ctl [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+		echo "Options:"
+		echo "	--no-reload : Do not reload the configuration and restart the container-management service automatically"
+		echo "	--ansi : Don't use colored output."
+		echo "	--verbose | -v : Enable verbose mode."
+		echo "	--help : This message."
+		echo
+		exit 0
+}
+
+while [ -n "$1" ]; do
```
Comment (mikehaller) [inline src/sh/sdv-kanto-ctl:111]: Should be using getopts

File: src/sh/sdv-kanto-ctl [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+         ($REGISTRY_HOSTNAME): {
+             "credentials": {
+                 "user_id": $REGISTRY_USERNAME,
+                 "password": $REGISTRY_PASSWORD
+             }
+         } }' < ${KANTO_CONFIG} > ${TMPFILE}
+	[ "$VERBOSE" = "1" ] && echo "Created temp file for modifications: ${TMPFILE}"
+
+	if [ -w ${KANTO_CONFIG} ]; then
+		if ! diff ${TMPFILE} ${KANTO_CONFIG} > /dev/null; then
```
Comment (mikehaller) [inline src/sh/sdv-kanto-ctl:265]: diff > 1 might mean an error, so better check the exit codes properly.

**Review by stlachev — APPROVED** [review]
LGTM

File: src/sh/sdv-kanto-ctl (hunk 2 above)
Comment (stlachev) [reply src/sh/sdv-kanto-ctl:265]: There is no operator >, Operator will be -gt I think

File: src/sh/sdv-kanto-ctl [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+	display_help
+	exit 0
+fi 
+
+[ "$COMMAND" = "add-registry" ] && add_registry
+[ "$COMMAND" = "remove-registry" ] && remove_registry
+[ "$COMMAND" = "remove-all-registries" ] && remove_all_registries
+[ "$COMMAND" = "list-registries" ] && list_registries
+[ "$COMMAND" = "show-config" ] && show_config
+[ "$COMMAND" = "set" ] && set_config
```
Comment (mikehaller) [inline src/sh/sdv-kanto-ctl:410]: Make commands more consistent, e.g.
"show-config" + "set-config"
vs.
"show" + "set"

File: src/sh/sdv-kanto-ctl [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	echo "Setting configuration $CONFIG_KEY to $CONFIG_VALUE"
+	TMPFILE=$(mktemp --tmpdir kanto-cm-config.XXXXXXX)
+	CONFIG_KEY=".${CONFIG_KEY}"
+	jq \
+	--arg CONFIG_KEY "$CONFIG_KEY" \
+	--arg CONFIG_VALUE "$CONFIG_VALUE" \
+	'setpath( $CONFIG_KEY|split(".")[1:]; $CONFIG_VALUE)' < ${KANTO_CONFIG} > ${TMPFILE}
+	[ "$VERBOSE" = "1" ] && echo "Created temp file for modifications: ${TMPFILE}"
+
+	if [ -w ${KANTO_CONFIG} ]; then
```
Comment (mikehaller) [inline src/sh/sdv-kanto-ctl:374]: Is there any way to make the Apply-logic more generic and as a function, instead of duplicating it?

**Review by mikehaller — COMMENTED** [review]
Check comments for requested changes

File: .github/workflows/build-leda-utils.yaml [hunk 5]
```diff
@@ -27,6 +27,10 @@ jobs:
         with:
           submodules: 'recursive'
           fetch-depth: 0
+      - name: Run ShellCheck
```
[the remark this answers is not in the document]
Comment (mikehaller) [reply .github/workflows/build-leda-utils.yaml:33]: Added shell checker checker

File: src/sh/sdv-kanto-ctl (hunk 3 above)
Comment (mikehaller) [reply src/sh/sdv-kanto-ctl]: Renamed command to "set-value" to make it more consistent

**Pull request merged**